Update links to JIRA tickets.

jonathanknowles opened the pull request:
## Issue

None

## Details

This PR updates all links to Jira tickets across the whole repository.

These are now hosted at `cardanofoundation.atlassian.net` instead of `input-output-hk.atlassian.net`.

jonathanknowles committed:
Update references to JIRA tickets.

These should now point to `cardanofoundation.atlassian.net`.

jonathanknowles committed:
Remove reference to "IOHK Jira".

HeinrichApfelmus commented on docs/site/src/design/concepts/eras.md:
```diff
@@ -16,7 +16,7 @@ The roadmap is actually divided into _phases_, under which one or more eras (har
 The mechanism whereby the Cardano Node operates in different modes - depending on which slot it is up to - is called the "Hard Fork Combinator".
 
-The following [seminars](https://input-output.atlassian.net/wiki/spaces/EN/pages/718962750/IOHK+Research+Engineering+Seminar) (company internal use only) are a good introduction:
+The following [seminars](https://cardanofoundation.atlassian.net/wiki/spaces/EN/pages/718962750/IOHK+Research+Engineering+Seminar) (company internal use only) are a good introduction:
```
```suggestion
The following [seminars](https://input-output.atlassian.net/wiki/spaces/EN/pages/718962750/IOHK+Research+Engineering+Seminar) (company internal use only) are a good introduction:
```

This is a link to the IOHK confluence, not our JIRA.

jonathanknowles replied:
Fixed in: 417815e821c477d826352d8a1654fb50898d2a9d.

HeinrichApfelmus reviewed (approved):
One link is wrong, but the others look good to me. 🤓

jonathanknowles committed:
Fix link to IOHK Confluence wiki page.

In response to code review:

https://github.com/cardano-foundation/cardano-wallet/pull/4077#discussion_r1280500182

Co-authored-by: Heinrich Apfelmus <heinrich.apfelmus@iohk.io>
